druid-shell/src/platform/x11/clipboard.rs

```diff
@@ -43,10 +89,107 @@ impl Clipboard {
         None
     }
 
-    pub fn get_format(&self, _format: FormatId) -> Option<Vec<u8>> {
-        // TODO(x11/clipboard): implement Clipboard::get_format
-        warn!("Clipboard::get_format is currently unimplemented for X11 platforms.");
-        None
+    pub fn get_format(&self, format: FormatId) -> Option<Vec<u8>> {
+        match self.get_format_impl(format) {
+            Ok(result) => result,
+            Err(error) => {
+                warn!("Error in Clipboard::get_format: {:?}", error);
+                None
+            }
+        }
+    }
+
+    fn get_format_impl(&self, format: FormatId) -> Result<Option<Vec<u8>>, ReplyOrIdError> {
+        let conn = &*self.connection;
+        let (format_atom, clipboard_atom, incr_atom) = {
+            let format = conn.intern_atom(false, format.as_bytes())?;
+            let clipboard = conn.intern_atom(false, b"CLIPBOARD")?;
+            let incr = conn.intern_atom(false, b"INCR")?;
+            (
+                format.reply()?.atom,
+                clipboard.reply()?.atom,
+                incr.reply()?.atom,
+            )
+        };
+
+        // Create a window for the transfer
+        let window = WindowContainer::new(conn, self.screen_num)?;
+
+        conn.convert_selection(
+            window.window,
+            clipboard_atom,
+            format_atom,
+            TRANSFER_ATOM,
+            self.timestamp.get(),
+        )?;
+
+        // Now wait for the selection notify event
+        conn.flush()?;
+        let notify = loop {
+            match conn.wait_for_event()? {
+                Event::SelectionNotify(notify) if notify.requestor == window.window => {
+                    break notify
+                }
+                event => self.event_queue.borrow_mut().push_back(event),
+            }
+        };
+
+        if notify.property == x11rb::NONE {
+            // Selection is empty
+            Ok(None)
```

Comment:
can early return

Reply:
Uhm, what do you mean? This `if` is the last expression in the function. This is the return value.

Do you want me to add an explicit `return` for clarity?

Reply:
Yes, this is the last expression. but using return here the code in the else block can be moved out. Also yes, explicit return looks clearer

Reply:
Okay, done.